### ISSUE TITLE
error getting docker info: No such file or directory

### ISSUE THREAD

**Issue by mindscratch** [influxdb/telegraf#84]
Running telegraf 0.1.4 on CentOS 7 with Docker 1.7.1 fails to gather docker metrics. What I get is

error getting docker info: open /sys/fs/cgroup/cpuacct/docker/<long hash>/cpuacct.stat: no such file or directory

I tried running `find / -name docker -type d` and only found:

/run/docker
/var/lib/docker
/etc/docker
/usr/libexec/docker

Perhaps the cgroup paths are different in CentOS/RHEL 7?

**Comment by sparrc** [comment]
@mindscratch can you please run `docker ps` and `docker version` ?

**Comment by mindscratch** [comment]
Docker 1.7.1

The output from docker ps isn't as each type get because it's on a separate
network (i can't just copy/paste).

**Comment by maxekman** [comment]
This also happens to me with boot2docker (OS X) and Docker v1.7.1.

**Comment by sparrc** [comment]
@maxekman, The docker plugin is not supported on OSX, because it is not possible to poll for docker information from the OSX, telegraf would have to be running from within the boot2docker VM for it to work.

I also realize that the supported plugins for various OSes need to be better documented, this is going to be a point of future work.

**Comment by maxekman** [comment]
Thanks for explaining. Then it will at least hopefully work in the deployment at CoreOS.


--

Max Ekman
Software Engineer

+46 708 710504
max@looplab.se

＞ looplab.se

**Comment by aheusingfeld** [comment]
@sparrc I seem to have the same problem on An AWS EC2 instance running docker 1.6.2! Is there any documentation on which volumes telegraf needs - might also be dependant on the OS, I assume

**Comment by sparrc** [comment]
@aheusingfeld This is something I'm looking into at the moment, unfortunately the docker functionality is completely based on the gopsutil package, which hasn't been the most stable option in terms of cross-platform functionality and build: https://github.com/shirou/gopsutil

**Comment by kenzodeluxe** [comment]
Same here:
$ cat /etc/issue
Ubuntu 15.04
$ uname -a
Linux kenzo-qa 3.19.0-26-generic #28-Ubuntu SMP Tue Aug 11 14:16:32 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
$ docker --version
Docker version 1.8.1, build d12ea79

**Comment by sparrc** [comment]
I believe that this PR: https://github.com/influxdb/telegraf/pull/180 will fix it, waiting on the author to sign the CLA and then will get this merged ASAP, thank you everyone for the issue reports

**Comment by sparrc** [comment]
I'm marking this as closed by PR #180, please re-open if that did not fix it
